### PR TITLE
*: migrate test-infra to testify for executor/prepared_test.go

### LINKED ISSUE #28603: migrate test-infra to testify for `prepared_test.go`

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #28603

Problem Summary:

### What is changed and how it works?

migrate test-infra to testify for prepared_test.go

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
@tisonkun @time-and-fate @qw4990 PTAL

**Review by qw4990 — APPROVED** [review]

**Comment by hawkingrei** [comment]
@time-and-fate @tisonkun @Reminiscent PTAL

**Review by rebelice — APPROVED** [review]

File: util/testutil/testutil.go [hunk 1]
```diff
@@ -342,6 +344,33 @@ func (t *TestData) GetTestCases(c *check.C, in interface{}, out interface{}) {
 	t.output[casesIdx].decodedOut = out
 }
 
+// GetTestCasesT gets the test cases for a test function.
+func (t *TestData) GetTestCasesT(tt *testing.T, in interface{}, out interface{}) {
```
Comment (time-and-fate) [inline util/testutil/testutil.go:348]: Can we reuse the methods in `testkit/testdata`?

Comment (hawkingrei) [reply util/testutil/testutil.go:348]: Yes, I will solve this problem in new PR.

**Pull request merged**